PermanentOrg/web-app
# PER-8345: Manage Tags/Archive Settings

## Description
This PR adds the manage tags functionality to the web-app, located in a new dialog called "Archive Settings", currently available to Archive Owners (and eventually Managers but the back end doesn't support that right now). This dialog lets you edit and delete existing tags in the current archive, allowing you to rename a tag on all records in the archive or delete an entire tag as well.

To hopefully help get this approved ASAP I'm going to preemptively explain a few of the specific changes made in this PR:
- Jest configuration is changed since the Manage-Tags tests are the first tests in the `CoreModule`, a specifically problematic module in our codebase for Jest Testing. The config changes help us ignore some errors from using some specific Typescript libraries that were not included in the previous Jest-compatible tests written in the `OnboardingModule`.
- The change to `src/app/notifications/services/notification.service.ts` is written for the same reason. Jest does not like the NodeJS.Timeout type. (But seriously, isn't this a browser application? Why are we using a builtin NodeJS type there anyway? Is that a thing you can just... do?)
- The introduction of the `TagVO` class in #63 requires the renaming of some methods that use the similarly shaped interface `TagVOData` but call them "TagVOs". You can see this renaming scheme in `src/app/shared/services/api/tag.repo.ts` but any files that call those methods are modified as well.
- The changes in `src/app/file-browser/components/edit-tags/edit-tags.component.ts` are to refresh the currently viewed item's tags when you update a tag in the new dialog. To test this out, select a record in My Files so you can see its metadata and tags list, then edit one of the tags in the Manage Tags dialog. With the changes listed, you should see the file's tag list instantly update instead of needing to be refreshed manually.
- I checked in a tiny amount of commented out code in `src/app/core/components/archive-settings-dialog/archive-settings-dialog.component.ts` because currently the backend does not allow managers to edit/delete tags, but that will be fixed eventually, and we can uncomment the code with that fix.

## Steps to Test
This functionality has been tested by QA already, but here's a thorough checklist:
- Open the Archive Settings and edit/delete tags
- Verify a confirmation box is shown when trying to delete a tag
- Verify the tag filter search thing works correctly
- Begin renaming a tag and then cancel out of editing it. The tag name *should* revert. Make sure to test this in conjunction with saving other tags.
- Click a record in My Files so you can see its metadata and tags list, and then open archive settings and update one of that record's tags. When the dialog closes, you should see the item's tag is updated immediately without the need to manually refresh.
- Test that you cannot use the Manage-Tags component without the proper access role.
- Test the Archive Settings dialog in mobile view.
- Make sure I didn't leave any `console.log`s lying around.
